lib/bq_stream/bq_stream.rb

```diff
@@ -46,7 +35,6 @@ def self.config_initialized
   end
 
   def self.initialize_old_records
-    # logger.info "#{Time.now}: $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$ Start Initialize Old Records $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$"
```

Comment:
So by commenting these out, even if you turn on logging it won't log anything.

I would uncomment these but add a method that will log if it is enabled. Also you can put all of these class methods inside of a `class << self` and remove the `self.`

```ruby
define_setting :batch_size, 1000

class << self
  attr_accessor :logger

  def self.log(type, message)
    return unless logger

    type = :info unless %i[unknown fatal error warn info debug].include?(type)

    logger.send(type, message)
  end
end
```